src/main/java/com/hillel/items_exchange/service/AdvertisementService.java

```diff
@@ -116,6 +119,26 @@ public void remove(long id) {
         advertisementRepository.flush();
     }
 
+    public void setAdvertisementDefaultImage(Long advertisementId, Long imageId, User owner) {
+        final Advertisement advertisement = owner.getAdvertisements().stream()
+                .filter(adv -> adv.getId() == advertisementId)
+                .findFirst().orElseThrow(() -> new IllegalIdentifierException(
+                        getExceptionMessageSourceWithAdditionalInfo(
+                                "exception.illegal.id",
+                                "User does not has advertisement with such id"
+                        )));
+        final Image image = advertisement.getProduct().getImages().stream()
+                .filter(img -> img.getId() == imageId)
+                .findFirst().orElseThrow(() -> new IllegalIdentifierException(
+                        getExceptionMessageSourceWithAdditionalInfo(
+                                "exception.illegal.id",
+                                "There is no image with such id in this advertisement"
+                        )));
+        advertisement.getProduct().getImages().forEach(img->img.setDefaultPhoto(false));
```

Comment:
I would create a method `Image getDefaultImage(Long advertisementId)`, then returned image.setDefaultPhoto(false). But it is only my thougts.